### PR TITLE
all: first cut of ce/fs2 3.0

### PR DESCRIPTION
**Pull request by ahjohannessen**
- adjust client / server runtime to CE/FS2 3.0
 - adjust code generation / tests accordingly.
 - change to munit

### CODE REVIEW

**Comment by ahjohannessen** [comment]
Just for early feedback

File: java-runtime/src/main/scala/client/Fs2UnaryClientCallListener.scala [hunk 1]
```diff
@@ -2,20 +2,25 @@ package org.lyranthe.fs2_grpc
...
 import cats.implicits._
 import io.grpc._
 
-class Fs2UnaryClientCallListener[F[_], Response](grpcStatus: Deferred[F, GrpcStatus], value: Ref[F, Option[Response]])(
-    implicit F: Effect[F]
-) extends ClientCall.Listener[Response] {
+class Fs2UnaryClientCallListener[F[_], Response](
+    grpcStatus: Deferred[F, GrpcStatus],
+    value: Ref[F, Option[Response]],
+    runner: UnsafeRunner[F]
```
Comment (rossabaker) [inline java-runtime/src/main/scala/client/Fs2UnaryClientCallListener.scala:15]: Is this the package-private `UnsafeRunner` in a public constructor?  Happens a couple more times.

**Review by rossabaker — COMMENTED** [review]
Thanks for grabbing this.  We'll need a branching strategy, assuming the cats-effect-2 version continues to be supported.

I'm not a cats-effect-3 expert yet, but this looks pretty reasonable. One question inline.

File: java-runtime/src/main/scala/client/Fs2UnaryClientCallListener.scala (hunk 1 above)
Comment (ahjohannessen) [reply java-runtime/src/main/scala/client/Fs2UnaryClientCallListener.scala:15]: `UnsafeRunner` is just for testing and only exposes the required surface area of `Dispatcher`. I will make sure to update constructors to package private or see if I can just stub `Dispatcher` in tests instead.

**Comment by ahjohannessen** [comment]
I thinking a `0.8.0` and a `0.7.x` might be sufficient?

File: java-runtime/src/main/scala/client/Fs2UnaryClientCallListener.scala (hunk 1 above)
Comment (ahjohannessen) [reply java-runtime/src/main/scala/client/Fs2UnaryClientCallListener.scala:15]: `Dispatcher` is `sealed` so I cannot stub it, I'll go with `UnsafeRunner` and make sure it does not leak.

**Comment by rossabaker** [comment]
Yeah, I'm good with bumping the minor version for this, and there can be a series/0.7.0 version for CE2.  That traps us if breaking changes are made in 0.7, which is only likely if sbt-protoc breaks, but perhaps by that point CE3 will be fairly well adopted.

An alternative is to start calling this 1.0.

**Comment by ahjohannessen** [comment]
>An alternative is to start calling this 1.0.

Or call it 0.9 and let CE2/ScalaPB have 0.8 ability for changes in ScalaPB 0.10.x -> 0.11.x ?

However, I am also good with 1.0.0-M for 2.13/dotty/ce/fs2 3.0/ScalaPB.0.11.x (support for dotty)

**Comment by rossabaker** [comment]
Let's go 1.0, since it's a big change for anyone using 0.x.  And Dotty could even still happen in the 0.x series, if that's ready before the CE3 ecosystem.  That's the strategy we've chosen on http4s.  Neither http4s upgrade is done yet, so it's unproven, but it looks good on paper! :smile:

**Review by rossabaker — APPROVED** [review]
:+1:, and we'll fork off a series branch from just before this merge.

**Comment by rossabaker** [comment]
Is this ready to go, then?

**Comment by ahjohannessen** [comment]
@rossabaker Sure. I will follow up with a PR that removes `UnsafeRunner` as `Dispatcher` is unsealed in latest ce3.

**Comment by rossabaker** [comment]
Okay, I forked off a series/0.x for #239.  I tried to rename `master` to `main` to catch up to modern usage, but we'll need @fiadliel to change the default branch.  But for now, master can be CE3.

**Comment by ahjohannessen** [comment]
:+1:

**Pull request merged**